### PR TITLE
Feature: draw area on map

### PR DESCRIPTION
**Pull request by edbrett**
Addresses the bridging of the create areas component to a generic form component alongside validation of fields and push notifications to the user:
- Form wrapper component to handle all form fields dynamically and validate against rules specified in /constants/validation-rules
- React-redux-toastr push notifications for failed validation
- Updated map component to pass features/layers from the map via props to its parents for integration for forms component
- Update areasCreate component to handle forms, initial onSubmit actions from react-validate, and convert all previous jsx to components worthy of rendering.

TODO:

- [ ] Create an action to POST the resulting JSON to the /areas endpoint of the forms API
- [ ] Abstract errors/notification component with react-redux-toastr to handle site wide notifications

### CODE REVIEW

File: public/index.html [hunk 1]
```diff
@@ -7,6 +7,8 @@
     <title>Forest Watcher</title>
     <base href="/" />
     <link href="https://fonts.googleapis.com/css?family=Open+Sans" rel="stylesheet">
+    <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/animate.css/3.5.2/animate.min.css">
+    <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/toastr.js/2.1.3/toastr.min.css">
```
Comment (sorodrigo) [inline public/index.html]: Could we import this via node_modules?

File: src/components/app/App.js [hunk 2]
```diff
@@ -55,6 +56,7 @@ class App extends React.Component {
           }
           {!user.loggedIn && <Redirect to="/" />}
         </main>
+        <ReduxToastr />
```
Comment (sorodrigo) [inline src/components/app/App.js]: I'm not sure this should be outside `<main>`, what do you think?

File: src/components/form/Form.js [hunk 3]
```diff
@@ -0,0 +1,26 @@
+import React from 'react';
+import Validation from 'react-validation';
+import ReactSelect from 'react-select';
+
+function withWrapper(Component) {
+  return class extends React.Component {
```
Comment (sorodrigo) [inline src/components/form/Form.js]: If there's no hooks involved, it's better to use a function.

File: public/index.html (hunk 1 above)
Comment (edbrett) [reply public/index.html]: indeed. done.

File: src/components/app/App.js (hunk 2 above)
Comment (edbrett) [reply src/components/app/App.js]: Not sure it matters so much but have done anyway as we dont want them displaying over the header.

File: src/components/form/Form.js (hunk 3 above)
Comment (edbrett) [reply src/components/form/Form.js]: This this is need as it is a wrapper component.

**Pull request merged**